### PR TITLE
Group version fix for RestAPI CoreV1

### PR DESCRIPTION
**Pull request by junshun**
Fixed RestClient Group Version for CoreV1 API

### CODE REVIEW

File: controllers/package_controller.go [hunk 1]
```diff
@@ -114,9 +115,11 @@ func RegisterPackageReconciler(mgr ctrl.Manager) (err error) {
 }
 
 func createSecretAuth(cfg *rest.Config) (auth.Authenticator, error) {
-	scheme := runtime.NewScheme()
-	cfg.GroupVersion = &api.GroupVersion
-	cfg.NegotiatedSerializer = serializer.NewCodecFactory(scheme)
```
Comment (TerryHowe) [inline controllers/package_controller.go:119]: This new approach has all sorts of hard coded strings. No better way?

Comment (junshun) [reply controllers/package_controller.go:119]: fixed

Comment (ewollesen) [reply controllers/package_controller.go:119]: I think the constants are OK, but FWIW, the k8s authors seem to be OK with using string literals in the instantiation of a GroupVersion. For example:
```sh
$ grep -r 'GroupVersion{Group: "", Version: "v1"}' | wc -l
914
```

Comment (ewollesen) [reply controllers/package_controller.go:119]: You might consider copying the cfg in this function too... It's kinda weird that a function named "createSecretAuth" would modify the argument you give it.

Comment (TerryHowe) [reply controllers/package_controller.go:119]: Whenever you start talking about the principle of least surprise, I think of the law of surprise from the Witcher. No first born children should b exchanged here.

I can live with the hard coded strings, I'm just a little surprised they don't have something better.

Comment (ewollesen) [reply controllers/package_controller.go:119]: > Whenever you start talking about the principle of least surprise, I think of the law of surprise from the Witcher. No first born children should b exchanged here.
> 
> I can live with the hard coded strings, I'm just a little surprised they don't have something better.

LOL. I love the Witcher! :) (And no one wants my first-born, I promise you! :P) 

I was surprised too, I thought they'd have some exported var ready to use or something... and to be clear, I really don't mean this as a counter to your suggestion Terry, just I noticed that Steven was following the pattern of the authors.

Comment (ewollesen) [reply controllers/package_controller.go:119]: But now I have to ask, did you find any surprises when you returned home from Seattle?

File: pkg/csset/csset_test.go [hunk 2]
```diff
@@ -29,7 +30,9 @@ func TestCSSetTwo(t *testing.T) {
 	set := NewCSSet("")
 	set.Add("one")
 	set.Add("two")
-	assert.Equal(t, "one,two", set.String())
+	exp := [2]string{"one", "two"}
```
Comment (ewollesen) [inline pkg/csset/csset_test.go:33]: If we end up using csset anywhere else, maybe we should write a test helper for this. Something that blends with the testify/assert package would be nice.

File: controllers/package_controller.go (hunk 1 above)
Comment (TerryHowe) [reply controllers/package_controller.go:119]: Ha! No surprises for me.

**Pull request merged**